pytorch_lightning/trainer/trainer.py

```diff
@@ -2042,9 +2042,13 @@ def devices(self) -> Optional[Union[List[int], str, int]]:
 
     @property
     def data_parallel_device_ids(self) -> Optional[List[int]]:
-        return (
-            self._accelerator_connector.parallel_device_ids if self._accelerator_connector.parallel_device_ids else None
-        )
+        rank_zero_deprecation("`Trainer.data_parallel_device_ids` was deprecated in v1.6.")
+        if isinstance(self.accelerator, GPUAccelerator):
+            if isinstance(self.strategy, ParallelStrategy):
+                return [i for i in range(len(self.strategy.parallel_devices))]
```

Comment:
Reviewers: here we can not use self.strategy.parallel_devices because self.strategy.parallel_devices could be a list of torch.devices

This is the existing logic, the correct one should be get ids from self.strategy.parallel_devices instead of `[i for i in range(len(self.strategy.parallel_devices))]`. Should we fix it? It will be a minor breaking change, and we have no active usage of this method (other than two gpu_log functions will be removed in 1.7)

Reply:
IMO yes

Reply:
Changed the logic to use device.index for each device in parallel_devices